instrumentation/opentelemetry-api-metrics-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/metrics/bridge/ApplicationBoundDoubleCounter.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.opentelemetryapi.metrics.bridge;
+
+import application.io.opentelemetry.api.metrics.BoundDoubleCounter;
+import application.io.opentelemetry.context.Context;
+import io.opentelemetry.javaagent.instrumentation.opentelemetryapi.context.AgentContextStorage;
+
+final class ApplicationBoundDoubleCounter implements BoundDoubleCounter {
+
+  private final io.opentelemetry.api.metrics.BoundDoubleCounter agentCounter;
+
+  ApplicationBoundDoubleCounter(io.opentelemetry.api.metrics.BoundDoubleCounter agentCounter) {
+    this.agentCounter = agentCounter;
+  }
+
+  @Override
+  public void add(double value) {
+    agentCounter.add(value);
+  }
+
+  @Override
+  public void add(double value, Context context) {
+    agentCounter.add(value, AgentContextStorage.getAgentContext(context));
```

Comment:
Interesting.  AgentContext needs to be wrapped in some fashion?  Does this not need to apply when looking up the implicit context in the normal "add" method?

Reply:
We bridge the actual context API too, so if a user made a context current in threadloc, it's actually an agent context that is made current, app contexts just wrap and don't have their own storage. Here we just unwrap if the user passed such a Context in explicitly.